File: src/main/java/org/dxworks/githubminer/dto/response/repository/pullrequests/PullRequest.java
package org.dxworks.githubminer.dto.response.repository.pullrequests;

import com.google.api.client.json.GenericJson;
import com.google.api.client.util.Key;
import lombok.Data;
import org.dxworks.githubminer.dto.commons.User;

@Data
public class PullRequest extends GenericJson {
    @Key
    private Long id;
    @Key
    private String state;
    @Key
    private String title;
    @Key
    private String body;
    @Key
    private User user;
    @Key
    private User assignee;
    @Key("merged_by")
    private User mergedBy;
    @Key("created_at")
    private String createdAt;
    @Key("merged_at")
    private String mergedAt;
    @Key("updated_at")
    private String updatedAt;
    @Key("closed_at")
    private String closedAt;
    @Key
    private PullRequestBranch head;
    @Key
    private PullRequestBranch base;
    @Key
    private Long number;
}

File: github-miner-core/src/main/kotlin/org/dxworks/githubminer/dto/response/repository/pullrequests/PullRequest.kt
package org.dxworks.githubminer.dto.response.repository.pullrequests

import com.google.api.client.json.GenericJson
import com.google.api.client.util.Key
import org.dxworks.githubminer.dto.commons.User

class PullRequest : GenericJson() {
    @Key
    var id: Number? = null

    @Key
    var state: String? = null

    @Key
    var title: String? = null

    @Key
    var body: String? = null

    @Key
    var user: User? = null

    @Key
    var assignee: User? = null

    @Key("merged_by")
    var mergedBy: User? = null

    @Key("created_at")
    var createdAt: String? = null

    @Key("merged_at")
    var mergedAt: String? = null

    @Key("updated_at")
    var updatedAt: String? = null

    @Key("closed_at")
    var closedAt: String? = null

    @Key
    var head: PullRequestBranch? = null

    @Key
    var base: PullRequestBranch? = null

    @Key
    var number: Number? = null
}
